sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/FileScanBuilder.scala

```diff
@@ -57,6 +64,27 @@ abstract class FileScanBuilder(
     StructType(fields)
   }
 
+  def pushFilters(partitionFilters: Seq[Expression], dataFilters: Seq[Expression]): Unit = {
+    this.partitionFilters = partitionFilters
+    this.dataFilters = dataFilters
+    this.pushedDataFilters = pushDataFilters(dataFilters)
```

Comment:
We can translate the data filters here before passing to `pushDataFilters`

Reply:
I feel the same way, and this can still allow us to have file data sources implementing `SupportsPushDownFilters`, correct? it feels weird that built-in data sources don't use the V2 API.

Reply:
Actually, we don't need to have file data sources implementing `SupportsPushDownFilters` any more. The purpose of `SupportsPushDownFilters` is to push filters (in the format of `Sources.Filter`) to file source. Now in this new `pushFilters(partitionFilters: Seq[Expression], dataFilters: Seq[Expression])`, we have already pushed both the partition filters and data filters to file source, so no need to implement `SupportsPushDownFilters` any more to push the filters.